torch/distributed/algorithms/ddp_comm_hooks/mixed_precision_hooks.py

```diff
@@ -14,8 +14,9 @@ class _AllreduceUpcastHookState:
     group, and a stream to run parameter and gradient upcasts.
     """
     ddp_weakref: Any
-    upcast_stream: torch.cuda.Stream
+    upcast_stream: torch.Stream
     wait_for_stream_enqueued: bool = False
+    device_module: Any = torch.cuda
```

Comment:
We shouldn't introduce `Any` to our typing annotations.
To address that, we should introduce a protocol that exposes the methods we need.

Reply:
> We shouldn't introduce `Any` to our typing annotations. To address that, we should introduce a protocol that exposes the methods we need.

yeah, I have used protocol to do it. @kumpera